app/controllers/export_controller.rb

```diff
@@ -57,6 +57,28 @@ def csv
     send_data csv_result, filename: "Sara-Alert-#{params[:workflow].capitalize}-#{params[:type].capitalize}-#{DateTime.now}.csv"
   end
 
+  def excel_comprehensive_patients
+    redirect_to(root_url) && return unless current_user.can_export?
+
+    # Grab patients to export
+    patients = current_user.viewable_patients.where(isolation: params[:workflow] == 'isolation')
+
+    # Do nothing if issue with request/permissions
+    redirect_to(root_url) && return if patients.nil?
```

Comment:
Would patients ever be nil? It looks like it is still just an ActiveRecord::Relation (even if it finds nothing).

Would suggest redirecting before line 64 if params[:workflow] is not isolation or exposure, or default to exposure. Then, if you still want to return if no patients, do a return if patients.size.positive? or empty?

Reply:
I agree with you in that I don't think it would ever be nil. I was just copying over the exact same code from the original csv export to make sure I wasn't changing any kind of logic (https://github.com/SaraAlert/SaraAlert/blob/master/app/controllers/export_controller.rb#L45)

I also think it would be a good idea to redirect if the workflow param is neither isolation nor exposure, but I think the current behavior right now for when there are no patients is to just continue with the export (which will just return the header rows and no patients). I think this makes sense because having no patients to export is not really an error like providing an invalid param is.

Reply:
I'm not sure what the reason (in the original csv export) for redirecting if patients is nil was for, but the comment for it is: 'Do nothing if issue with request/permissions', but I think the permissions is handled with this already, which comes beforehand:
`redirect_to(root_url) && return unless current_user.can_export?`

I can go ahead and change the logic for the csv export as well

Reply:
Go ahead and change the logic in both places; An empty XLS makes sense for the second point.